packages/flutter_tools/templates/app/macos.tmpl/Runner.xcodeproj/project.pbxproj.tmpl

```diff
@@ -280,7 +280,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "\"$FLUTTER_ROOT\"/packages/flutter_tools/bin/macos_assemble.sh\ntouch Flutter/ephemeral/tripwire\n";
+			shellScript = "\"$FLUTTER_ROOT\"/packages/flutter_tools/bin/macos_assemble.sh";
```

Comment:
A reasonable follow up to this PR would be to add an automatic migration using the tooling that @jmagman already wrote, but its not something I have time for right now

Reply:
I'm guessing it doesn't matter if it gets "tripped" twice in a row?

Reply:
No, just as long as it gets modified at least once

Reply:
Isn't the reason that we were doing this here was that if we fix it to not need the tripwire, we need to update the project with the input/output files? It seems like this sets us up to accidentally change the script later while forgetting that we need to fix all the projects in the wild.

Reply:
The project input/output files are correct and can work 100% of the time, if you are building through flutter. When building through the xcode UI, changing configurations does not lead to any sort of change in files on disk - so Xcode thinks that our task can be skipped.

I have no idea how to fix this.

Reply:
> The project input/output files are correct

Right, but we took them out of the Xcode project when adding the tripwire. A fixed build flow that uses the input/output files and doesn't need a tripwire requires adding those back to the project. There's no world where we can fix the need for the tripwire without changing people's existing projects, so I don't see why we would want to move the tripwire creation from the project to the script.

Reply:
> changing configurations does not lead to any sort of change in files on disk - so Xcode thinks that our task can be skipped.
> 
> I have no idea how to fix this.

I thought we had a potential solution for this now, inspired by @jmagman's iOS work with App.framework: have three different copies of the framework in ephemeral, rather than just one (e.g., ephemeral/debug/..., ephemeral/profile/..., ephemeral/release/...) and control the linking ourselves with linker flags, including a configuration-variable-based linker search directory.

Reply:
different locations for debug/profile/release artifacts would definitely help. I can't say for sure if it would solve the issue completely

Reply:
Anyway, that part was a tangent. Going back to my main concern: what is the goal for moving this logic out of the project? This PR seems unrelated to the tripwire, and AFAIK nothing has changed since we made a deliberate decision to put the tripwire logic on this side of the divide (per my comment above).

Reply:
I think (now I need to double check, because I could have mixed it up) that ending with `&& touch` was causing failures of the actual script to be suppressed. I bet there is some sort of bash-fu to fix it..

Reply:
Oh, I see. I missed that there wasn't anything else in macos_assemble.sh before, and thought the error exit you added there was to fix that problem.

Yeah, the current Xcode script is definitely wrong, because it's just two statements. Changing the `\n` to ` && ` should be the easy fix there.